internal/k8shandler/forwarding.go

```diff
@@ -39,59 +40,69 @@ func EvaluateAnnotationsForEnabledCapabilities(forwarder *logging.ClusterLogForw
 }
 
 func (clusterRequest *ClusterLoggingRequest) generateCollectorConfig() (config string, err error) {
-	if clusterRequest.Cluster == nil || clusterRequest.Cluster.Spec.Collection == nil {
-		log.V(2).Info("skipping collection config generation as 'collection' section is not specified in CLO's CR")
-		return "", nil
-	}
-	switch clusterRequest.Cluster.Spec.Collection.Type {
-	case logging.LogCollectionTypeFluentd:
-		break
-	case logging.LogCollectionTypeVector:
-		break
-	default:
-		return "", fmt.Errorf("%s collector does not support pipelines feature", clusterRequest.Cluster.Spec.Collection.Type)
-	}
 
-	if clusterRequest.Forwarder == nil {
-		clusterRequest.Forwarder = &logging.ClusterLogForwarder{}
+	var collectionSpec *logging.CollectionSpec
```

Comment:
All of these should happen earlier in our reconciliation loop.  We should be able to push everything up to where the "ClusterLoggingRequest" is initialized so that all subsequent usages already have what is needed.  Generally:

* watch callback
* load resource
* call k8shandler function with what we have... it may need to receive a resource we created